lib/tapioca/commands/gem.rb

```diff
@@ -128,6 +136,26 @@ def sync(should_verify: false)
 
       private
 
+      sig { params(gem_queue: T::Array[Tapioca::Gemfile::GemSpec]).void }
+      def verify_rbi_contents(gem_queue)
+        result = Executor.new(gem_queue, number_of_workers: @number_of_workers).run_in_parallel do |gem|
+          rbi_string = compile_gem_rbi(gem, silent: true).sub(/^# typed: .*/, "")
+          file = T.unsafe(Pathname).glob((@outpath / "#{gem.name}@*.rbi").to_s).first
+          path = gem_rbi_filename(gem.name, gem.version)
+
+          if file.nil?
+            [path, :added]
+          elsif file.basename.to_s != gem.rbi_file_name || File.read(file).sub(/^# typed: .*/, "") != rbi_string
```

Comment:
This is surprising I'd expect both runs to use the same `typed_overrides` and `auto_strictness` options.

Are we sure the diff is not because of the broken `main` we have right now?

Reply:
Yes. `compile_gem_rbi` returns the compiled rbi string before type checking and downgrading the type sigil, which means that if the RBI on disk has been downgraded, it will always fail unless we ignore the sigil part.